hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/JavaClassNameResolver.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.keyvalue;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public final class JavaClassNameResolver {
+
+    private static final Map<String, String> CLASS_NAMES_BY_FORMAT = new HashMap<String, String>() {{
```

Comment:
We may rather use `ImmutableMap` rather than subclass, we depend on guava anyway due to calcite.